presto-main/src/main/java/com/facebook/presto/catalogserver/RemoteMetadataManager.java

```diff
@@ -185,6 +148,71 @@ public List<QualifiedObjectName> getReferencedMaterializedViews(Session session,
                 : readValue(referencedMaterializedViewsListJson, new TypeReference<List<QualifiedObjectName>>() {});
     }
 
+    @Override
+    public MetadataResolver getMetadataResolver(Session session)
```

Comment:
a non-public outer class would work well here to avoid bloat.

Reply:
Or perhaps there's no extra class at all. Instead RemoteMetadataManager simply implements MetadataResolver.